core/src/main/java/io/grpc/internal/AbstractServerStream.java

```diff
@@ -243,7 +267,19 @@ public final void transportReportStatus(Status status) {
      * #transportReportStatus}.
      */
     public void complete() {
-      closeListener(Status.OK);
+      if (deframerClosed) {
+        deframerClosedTask = null;
+        closeListener(Status.OK);
+      } else {
+        deframerClosedTask =
+            new Runnable() {
+              @Override
+              public void run() {
+                closeListener(Status.OK);
+              }
+            };
+        closeDeframer(false);
```

Comment:
This seems like it should be `true` as well to match the previous code. Generally, on the server-side we don't continue deframing after we have a status. This seems like it could cause a deadlock (in that we make no further progress). I'd have sort of expected a test to catch this... maybe we're missing a test case here.